plugin/src/main/java/io/fabric8/maven/plugin/converter/DeploymentOpenShiftConverter.java

```diff
@@ -86,10 +86,18 @@ public HasMetadata convert(HasMetadata item) {
                 if (strategy != null) {
                     strategyType = strategy.getType();
                 }
-                if (openshiftDeployTimeoutSeconds != null && openshiftDeployTimeoutSeconds > 0 &&
-                        (Strings.isNullOrBlank(strategyType) || "Rolling".equals(strategyType))) {
-                    specBuilder.withNewStrategy().withType("Rolling").
-                        withNewRollingParams().withTimeoutSeconds(openshiftDeployTimeoutSeconds).endRollingParams().endStrategy();
+                if (openshiftDeployTimeoutSeconds != null && openshiftDeployTimeoutSeconds > 0) {
+                    if (Strings.isNullOrBlank(strategyType) || "Rolling".equals(strategyType)) {
+                        specBuilder.withNewStrategy().withType("Rolling").
+                            withNewRollingParams().withTimeoutSeconds(openshiftDeployTimeoutSeconds).endRollingParams().endStrategy();
+                    } else if (Strings.isNotBlank(strategyType)) {
```

Comment:
I just pushed a fix that tries to make the code look identical for `Rolling` and `Recreate` (other than we default to `Rolling` if no strategy is yet specified). 

Unfortunately the sub-types in kubernetes have no real polymorphism OOTB (e.g. the `Custom` strategy has no `timeoutSeconds` unlike `Rolling` and `Recreate`) - its common for them to be very different too (e.g. Volume types). Nor do we yet have polymorphic interfaces & builder APIs generated yet; that requires a much larger amount of work to be done on kubernetes-model which is outside of the fix for this issue ;)